### PR TITLE
Port old conrod example "simple_ui" to egui

### LINKED ISSUE #856: Port old `simple_ui.rs` example to use `nannou_egui`
Moved to the following location in #855 as it uses conrod:

https://github.com/nannou-org/nannou_conrod/blob/main/examples/simple_ui.rs

It would be nice to port this back to this repo under `examples/ui/egui/simple_ui.rs` using the `nannou_egui` crate introduced in #854.

### PR DESCRIPTION
**Pull request by jorikvanveen**
This PR will add a new example: `examples/ui/egui/simple_ui.rs`, it's the egui version of an old example found here: https://github.com/nannou-org/nannou_conrod/blob/main/examples/simple_ui.rs. As requested in #856.

I'm relatively new to nannou and this is my first contribution so don't be afraid to tell me how I can do better!

Closes #856

### CODE REVIEW

**Comment by jorikvanveen** [comment]
Okay everything works except the position variable, the old example used an XYPad for this, but I can't really find anything comparable. I could implement something like XYPad myself, but that would complicate the example and I'm not really sure if that's worth it considering the name of the example.

File: examples/ui/egui/simple_ui.rs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+fn update(_app: &App, model: &mut Model, update: Update) {
+    let egui = &mut model.egui;
+    let settings = &mut model.settings;
+
+    egui.set_elapsed_time(update.since_start);
+    let ctx = egui.begin_frame();
+
+    egui::Window::new("Settings").show(&ctx, |ui| {
+        // Resolution slider
+        ui.add(egui::Label::new("Resolution:"));
```
Comment (mitchmindtree) [inline examples/ui/egui/simple_ui.rs:54]:
```suggestion
        ui.label("Resolution:");
```

File: examples/ui/egui/simple_ui.rs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        ui.add(egui::Slider::new(&mut settings.resolution, 1..=40));
+
+        // Scale slider
+        ui.add(egui::Label::new("Scale:"));
```
Comment (mitchmindtree) [inline examples/ui/egui/simple_ui.rs:58]:
```suggestion
        ui.label("Scale:");
```

File: examples/ui/egui/simple_ui.rs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        ui.add(egui::Slider::new(&mut settings.scale, 0.0..=1000.0));
+
+        // Rotation slider
+        ui.add(egui::Label::new("Rotation:"));
```
Comment (mitchmindtree) [inline examples/ui/egui/simple_ui.rs:62]:
```suggestion
        ui.label("Rotation:");
```

File: examples/ui/egui/simple_ui.rs [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        ui.add(egui::Slider::new(&mut settings.rotation, 0.0..=360.0));
+
+        // Random color button
+        let clicked = ui.add(egui::Button::new("Random color")).clicked();
```
Comment (mitchmindtree) [inline examples/ui/egui/simple_ui.rs:66]:
```suggestion
        let clicked = ui.button("Random color").clicked();
```

File: examples/ui/egui/simple_ui.rs [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+        if clicked {
+            settings.color = rgb(random(), random(), random());
```
Comment (mitchmindtree) [inline examples/ui/egui/simple_ui.rs:69]: Out of curiosity, does this work?
```suggestion
            settings.color = random();
```

File: examples/ui/egui/simple_ui.rs [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    model.egui.handle_raw_event(event);
+}
+
+fn view(app: &App, model: &Model, frame: Frame) {
+    let settings = &model.settings;
+
+    let draw = app.draw();
+    draw.background().color(BLACK);
+
+    let rotation_radians = settings.rotation / 360.0 * 2.0 * std::f32::consts::PI;
```
Comment (mitchmindtree) [inline examples/ui/egui/simple_ui.rs:85]:
```suggestion
    let rotation_radians = deg_to_rad(settings.rotation);
```

**Review by mitchmindtree — COMMENTED** [review]
Thanks so much @jorikvanveen, looking good!

I've left a few small suggestions, if you could address these I'd be happy to land this :)

File: examples/ui/egui/simple_ui.rs (hunk 5 above)
Comment (jorikvanveen) [reply examples/ui/egui/simple_ui.rs:69]: It doesn't, unfortunately, it gives this error:
image

**Comment by jorikvanveen** [comment]
Okay I just committed everything (except the random color thing), LGTM!

**Pull request merged**